qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -16,12 +16,12 @@
 
 from abc import ABC, abstractmethod
 from collections.abc import Collection, Mapping
-from numbers import Number
-from typing import Iterator, Sequence
+from numbers import Complex
```

Comment:
Why did you change this to `Complex`? I chose `Number` on purpose to leave this as general as possible.

Reply:
Changing it to `Number` gives these mypy errors:
```python
qiskit_nature/second_q/operators/sparse_label_op.py: note: In member "_multiply" of class "SparseLabelOp":
qiskit_nature/second_q/operators/sparse_label_op.py:227: error: Unsupported operand types for * ("complex" and "Number")  [operator]
qiskit_nature/second_q/operators/sparse_label_op.py:227: note: Both left and right operands are unions
```

Reply:
Looks like a compatibility limitation of the numbering stack and typing.
I suggest you use `# type: ignore[operator]` but leave the type to be `Number` as originally intended.

Reply:
I think `Complex` is actually correct. The error
```
Unsupported operand types for * ("complex" and "Number")
```
explains why. Having `Number` there was a bug that slipped through mypy because the type was incorrect in the function signature (it had `complex`, so mypy assumed `other` was `complex`.
https://github.com/Qiskit/qiskit-nature/blob/ba3a191b30606d6c1da5c378de2eea7bceaeab57/qiskit_nature/second_q/operators/sparse_label_op.py#L194

Reply:
@mrossinek I tried replacing everything but that gave other errors. I think it is a bit delicate. In any case it's unrelated to the rest of this PR so I think it's better to do it in a separate PR. I opened https://github.com/Qiskit/qiskit-nature/issues/896 for it.